tests/test_model.py

```diff
@@ -218,18 +218,29 @@ def test_pickle(self):
         num_model = CallableNumericalModel(
             {y: a * x ** b}, independent_vars=[x], params=[a, b]
         )
+        connected_num_model = CallableNumericalModel(
+            {y: a * x ** b}, connectivity_mapping={y: {x, a, b}}
+        )
         # Test if lsoda args and kwargs are pickled too
         ode_model = ODEModel({D(y, x): a * x + b}, {x: 0.0}, 3, 4, some_kwarg=True)
-        for model in [exact_model, constraint, num_model, ode_model]:
+
+        models = [exact_model, constraint, num_model, ode_model,
+                  connected_num_model]
+        for model in models:
             new_model = pickle.loads(pickle.dumps(model))
             # Compare signatures
             self.assertEqual(model.__signature__, new_model.__signature__)
             # Trigger the cached vars.
-            model.vars, model.connectivity_mapping
-            new_model.vars, new_model.connectivity_mapping
+            model.vars
+            new_model.vars
```

Comment:
Add a comment that we trigger the cached attributes *because we compare `__dict__` s*